### ISSUE TITLE
getAssignedPartitions() returns null

### ISSUE THREAD

**Issue by chmielewskitomasz** [spring-projects/spring-kafka#1145]
While connecting to kafka I'm getting

```
Error while processing: null

java.lang.NullPointerException: null
	at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.seekPartitions(KafkaMessageListenerContainer.java:636)
	at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.run(KafkaMessageListenerContainer.java:764)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

there is 20 partitions, I'm connecting to kafka with two instances of application.

Is this because the first instance locks all the partitions?
It seems that when topic becomes empty problem disappears

**Comment by garyrussell** [comment]
It's a bug; looks like there is a race condition; we shouldn't be calling that code until some partitions have been assigned.

It's fixed in 2.2.x (currently 2.2.7) but we need to back-port the fix to earlier versions.

**Comment by chmielewskitomasz** [comment]
We're using `org.springframework.kafka:spring-kafka:2.2.0.RELEASE`

**Comment by garyrussell** [comment]
It's best to keep up to date with releases to get bug fixes like this; it was fixed in 2.2.1; the current 2.2.x version is 2.2.8.

**Comment by garyrussell** [comment]
It's best to keep up to date with releases to get bug fixes like this; it was fixed in 2.2.1; the current 2.2.x version is 2.2.7.

https://github.com/spring-projects/spring-kafka/commit/04e9e1a

**Comment by artembilan** [comment]
Fix via: https://github.com/spring-projects/spring-kafka/pull/1147
